src/lib/configure_dynatrace.py

```diff
@@ -0,0 +1,123 @@
+import asyncio
+import re
+import os
+from os import listdir
+from os.path import isfile
+from typing import NamedTuple, List, Optional
+from urllib.parse import urljoin
+
+from aiohttp import ClientSession
+
+from lib.context import LoggingContext
+
+from lib.fast_check import get_dynatrace_token_metadata
+from lib.credentials import fetch_dynatrace_url, fetch_dynatrace_api_key
+import yaml
+import json
+from main import is_yaml_file
+
+class ConfigureDynatrace:
+    async def post_dashboard(self, dynatrace_url: str, dynatrace_api_key: str, path: str, name:str, timeout: Optional[int] = 2) -> dict:
+        try:
+            with open(path, encoding="utf-8") as dashboard_file:
+                 dashboard_json = json.load(dashboard_file) 
+
+            response = await self.session.post(
+                url=urljoin(dynatrace_url, "/api/config/v1/dashboards"),
+                headers={
+                    "Authorization": f"Api-Token {dynatrace_api_key}",
+                    "Content-Type": "application/json; charset=utf-8"
+                },
+                timeout=timeout,
+                json=dashboard_json)
+            if response.status != 201:
+                response_json=  await response.json()
+                self.logging_context.log(f'Unable to create dashboard {name} in Dynatrace: {response.status}, url: {response.url}, reason: {response.reason}, message {response_json}')                                
+                return {}
+            else:
+                self.logging_context.log(f"Installed dashboard {name}")
+            return {}
+        except Exception as e:
+            self.logging_context.log(f'Unable to create dashboard in Dynatrace. Error details: {e}')
+            return {}
+
+    async def get_existing_dashboards(self, dynatrace_url: str, dynatrace_api_key: str, timeout: Optional[int] = 2) -> dict:
+        try:
+            response = await self.session.get(
+                url=urljoin(dynatrace_url, "/api/config/v1/dashboards"),
+                headers={
+                    "Authorization": f"Api-Token {dynatrace_api_key}",
+                    "Content-Type": "application/json; charset=utf-8"
+                },
+                timeout=timeout)
+            if response.status != 200:
+                self.logging_context.log(f'Unable to get existing dashboards config: {response.status}, url: {response.url}, reason: {response.reason}')
+                return {}
+
+            dashboards_json=  await response.json()
+            all_dashboards = [dashboard["name"] for dashboard in dashboards_json.get("dashboards",[])]
+            return all_dashboards
+        except Exception as e:
+            self.logging_context.log(f'Unable to get existing dashboards config. Error details: {e}')
+            return {}
+
+
+    async def get_available_dashboards(self) -> dict:

```

Comment:
Please change return value type to List and fix line 96 - we are returning different type on successful (List of dicts) and failed (empty dict) call.